File: ios/Plugin/NativeGeocoderPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(NativeGeocoderPlugin, "NativeGeocoder",
           CAP_PLUGIN_METHOD(reverseGeocode, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(forwardGeocode, CAPPluginReturnPromise);
)

File: ios/Plugin/NativeGeocoderPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(NativeGeocoderPlugin)
public class NativeGeocoderPlugin: CAPPlugin {
    private let implementation = NativeGeocoder()

    @objc func reverseGeocode(_ call: CAPPluginCall) {
        guard let latitude = call.getDouble("latitude") else {
            call.reject("Missing latitude")
            return
        }
        guard let longitude = call.getDouble("longitude") else {
            call.reject("Missing longitude")
            return
        }
        implementation.reverseGeocode(latitude: latitude, longitude: longitude, call: call)
    }

    @objc func forwardGeocode(_ call: CAPPluginCall) {
        guard let addressString = call.getString("addressString") else {
            call.reject("Missing addressString")
            return
        }
        implementation.forwardGeocode(address: addressString, call: call)
    }
}
